src/api-binder.coffee

```diff
@@ -263,6 +277,34 @@ module.exports = class APIBinder
 				body: updatedFields
 			.timeout(conf.apiTimeout)
 
+	pinDevice: ({ app, commit }) =>
+		@config.get('deviceId')
+		.then (deviceId) =>
+			@resinApi.get
+				resource: 'release'
+				options:
+					filter:
+						belongs_to__application: app
+						commit: commit
+						status: 'success'
+					select: 'id'
+			.then (release) =>
+				releaseId = _.get(release, '[0].id')
+				if !releaseId?
+					throw new Error('Cannot continue pinning preloaded device! No release found!')
+				@resinApi.patch
+					resource: 'device'
+					id: deviceId
+					body:
+						should_be_running__release: releaseId
+			.then ->
```

Comment:
Should be fat arrow

Reply:
I hate coffeescript... Good catch. I'll test it all again tomorrow before merge.